tools/toolchains/arm.py

```diff
@@ -413,6 +413,8 @@ def __init__(self, target, *args, **kwargs):
             self.flags['common'].append("-mfloat-abi=softfp")
         elif target.core.startswith("Cortex-M23"):
             self.flags['common'].append("-march=armv8-m.base")
+        elif target.core.startswith("Cortex-M33"):
```

Comment:
Cortex-M33 CPU specific flags are covered here.. https://github.com/ARMmbed/mbed-os/pull/8449/files#diff-9cf75aea879b6b9cc5026af0b8f708a3R396 

No need of `arch` flag for M33 as we give add `cpu` flag, we should be just adding floating point here for `F` option.

Just FYI- Special case of non-floating point related is covered here for Cortex-M23 since we do not specify CPU but arch only. Can still be moved here https://github.com/ARMmbed/mbed-os/pull/8449/files#diff-9cf75aea879b6b9cc5026af0b8f708a3R403.

Reply:
`-mcpu=cortex-m33+nodsp and -mfpu=none` Is already set, so addition of `-march` is not needed here. Code should be updated to:
```
elif target.core.startswith("Cortex-M33F"): 
    self.flags['common'].append("-mfpu=fpv5-sp-d16")
    self.flags['common'].append("-mfloat-abi=softfp")
```